src/altinn-app-frontend/src/components/presentation/NavBar.test.tsx

```diff
@@ -1,26 +1,69 @@
-import { mount } from 'enzyme';
 import React from 'react';
-import NavBar from './NavBar';
+import NavBar, { INavBarProps } from './NavBar';
+import { screen } from '@testing-library/react';
+import { getLanguageFromCode } from 'altinn-shared/language';
+import userEvent from '@testing-library/user-event';
+import { IAppLanguage } from 'altinn-shared/types';
+import { renderWithProviders } from '../../../testUtils';
+
+const renderNavBar = (props?: Partial<INavBarProps>) => {
+  const mockClose = jest.fn();
+  const mockBack = jest.fn();
+  const mockAppLanguageChange = jest.fn();
+  const mockAppLanguages: IAppLanguage[] = [
+    {
+      language: 'nb',
+    },
+    {
+      language: 'en',
+    },
+  ];
+  renderWithProviders(
+    <NavBar
+      language={getLanguageFromCode('nb')}
+      handleClose={mockClose}
+      handleBack={mockBack}
+      appLanguages={mockAppLanguages}
+      onAppLanguageChange={mockAppLanguageChange}
+      selectedAppLanguage={'nb'}
+      {...props}
+    />,
+  );
+
+  return { mockClose, mockBack, mockAppLanguageChange };
+};
 
 describe('components/presentation/NavBar.tsx', () => {
-  it('should render back button by default', () => {
-    const wrapper = mount(
-      <NavBar language={{}} handleClose={null} handleBack={null} />,
-    );
-    expect(wrapper.find('div.a-modal-navbar')).toHaveLength(1);
-    expect(wrapper.find('button.a-modal-close')).toHaveLength(1);
+  it('should render close button', async () => {
+    const { mockClose } = renderNavBar();
+    const closeButton = screen.getByRole('button', { name: /Lukk Skjema/i });
+    // Should not render back button
+    expect(screen.queryAllByRole('button')).toHaveLength(1);
```

Comment:
This check isn't cohesive as the context is rendering the close button.
Good thing that you have a comment on top, but I would extract it into a separate test.
I would consider using the `data-testid='uniqueId'` attribute on the component.
That way you actually can check for it with a query.
Not that big of a deal though.

Reply:
I changed to test id for the back button and updated the app language saga to fetch only if showLanguageSelector exists and is set to true in your layoutsettingsfile. The frontend will work as normal (not displaying an error) if you havent configured app language yet.